### PR TITLE
Add integration tests for the Permissions class

### PR DESCRIPTION
**Pull request by acicovic**
## Description
This PR adds some integration tests for the `Permissions` class, notably for the `get_user_roles_with_edit_posts_cap()` and `current_user_can_use_pch_feature()` functions, which are the main functions responsible for access to PCH features.

## Motivation and context
Cover more of our code, especially critical parts of it like access control.

## How has this been tested?
The introduced tests pass.

## Summary by CodeRabbit

- **Tests**
	- Introduced a comprehensive integration test suite for user role permissions, ensuring proper functionality in editing posts and accessing Content Helper features.
	- Added specific tests to validate user roles and their capabilities related to content permissions.
	- Enhanced reliability of the Parsely framework by covering various edge cases in access control.

### CODE REVIEW

File: tests/Integration/PermissionsTest.php [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+		);
+
+		// Filter set to true.
+		add_filter( 'wp_parsely_current_user_can_use_pch_feature', '__return_true' );
+			self::assertFalse(
+				Permissions::current_user_can_use_pch_feature(
+					$feature,
+					$pch_options // @phpstan-ignore-line
+				)
+			);
```
Comment (vaurdan) [inline tests/Integration/PermissionsTest.php]: I believe the code indentation here is a bit offset.
```suggestion
		self::assertFalse(
			Permissions::current_user_can_use_pch_feature(
				$feature,
				$pch_options // @phpstan-ignore-line
			)
		);
```

**Review by vaurdan — APPROVED** [review]
Looks great! I have added a minor comment, but approved nonetheless.

File: tests/Integration/PermissionsTest.php (hunk 1 above)
Comment (acicovic) [reply tests/Integration/PermissionsTest.php]: Good catch!
